crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -557,8 +555,11 @@ def save_upgrade_state(message = "")
       end
 
       def save_error_state(message = "")
```

Comment:
What if we'd just raise an exception (with `message`) from `save_error_state`, catch it in `api/upgrade.rb` (nodes method) and call `end_step` once we catch the exception?

Than we could remove all those checks for `false` when calling methods from `api/node.rb` ...